lib/puppet/provider/group/gpasswd.rb

```diff
@@ -0,0 +1,75 @@
+require 'puppet/provider/group/groupadd'
+
+Puppet::Type.type(:group).provide :gpasswd, :parent => Puppet::Type::Group::ProviderGroupadd do
+  require 'set'
+  require 'shellwords'
+
+  desc <<-EOM
+    Group management via `gpasswd`. This allows for local group
+    management when the users exist in a remote system.
+  EOM
+
+  commands  :addmember => 'gpasswd',
+            :delmember => 'gpasswd'
+
+  has_feature :manages_members unless %w{HP-UX Solaris}.include? Facter.value(:operatingsystem)
```

Comment:
Well, this is that sticky place where just confining it to gpasswd is never going to be right as OS's are added and keeping a list up to date is also never going to be complete.

BUT, as of now, it works on Linux systems that I know of. That code was copied from one of the other modules in that set so this may be something that needs to be changed in multiple locations.